crates/core/src/nats/streams.rs

```diff
@@ -0,0 +1,189 @@
+use std::collections::HashMap;
+
+use futures_util::future::try_join_all;
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+
+use super::{
+    client::NatsClient,
+    types::{
+        JetStreamConfig,
+        NatsConsumer,
+        NatsStorageType,
+        NatsStream,
+        PullConsumerConfig,
+    },
+    NatsError,
+    SubjectName,
+};
+
+#[derive(Debug, EnumIter, Clone, Hash, Eq, PartialEq)]
+pub enum StreamKind {
+    Blocks,
+    Transactions,
+}
```

Comment:
I am not sure why we decided to separate `StreamKind` from `SubjectName`. The complexity grows significantly doing this.

Reply:
A better abstraction might be to get merge of `Subjects` & `StreamKind` by creating an Enum type for each Stream:

`pub enum BlocksStream { Blocks }`

`pub enum TransactionsStream { Transactions, TransactionsById {...} }`

What do you think?